src/server/controllers/templateController.js

```diff
@@ -0,0 +1,102 @@
+const Quiz = require('../models/quiz');
+const QuizTemplate = require('../models/templates');
+
+const fetchTemplates = async (req, res) => {
+  try {
+    const templates = await QuizTemplate.find();
+    return res.json(templates);
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({ error: 'Failed to fetch templates' });
+  }
+};
+
+const createTemplate = async (req, res) => {
+  try {
+    const userId = req.user.id;
+    const { name, description, questions, duration } = req.body;
+
+    const template = new QuizTemplate({
+      name,
+      description,
+      questions,
+      duration,
+      createdBy: userId,
+    });
+
+    await template.save();
+    return res
+      .status(201)
+      .json({ message: 'Template created successfully', template });
+  } catch (error) {
+    console.log(error);
+    return res
+      .status(500)
+      .json({ error: 'Server error while creating template' });
+  }
+};
+
+const cloneTemplate = async (req, res) => {
```

Comment:
What is the purpose of cloning ? How are you planning to integrate ?

Reply:
The current implementation implements the creation of new quizzes from pre-existing templates.
Do you require something else or is something else expected of it ?
A little briefing would be greatly helpful